routing/datasource_test.go

```diff
@@ -126,6 +127,65 @@ func TestProcessRouteDefErrors(t *testing.T) {
 	}
 }
 
+func TestProcessRouteDefWeight(t *testing.T) {
+
+	wps := weightedPredicateSpec{}
+
+	cpm := make(map[string]routing.PredicateSpec)
+	cpm[wps.Name()] = wps
+
+	for _, ti := range []struct {
+		route  string
+		weight int
+	}{
+		{
+			`Path("/foo") -> <shunt>`,
+			0,
+		}, {
+			`WeightedPredicate10() -> <shunt>`,
+			10,
+		}, {
+			`Weight(20) -> <shunt>`,
```

Comment:
Please add also a test with more than one `Weight` to make sure the count is like we want to have.

Reply:
Done.